garak/generators/openai.py

```diff
@@ -229,10 +217,31 @@ def _load_client(self):
             r"^.+-[01][0-9][0-3][0-9]$", self.name
         ):  # handle model names -MMDDish suffix
             self.generator = self.client.completions
+        else:
+            raise ValueError(
+                f"No {self.generator_family_name} API defined for '{self.name}' in generators/openai.py - please add one!"
+            )
 
     def _clear_client(self):
         self.generator = None
         self.client = None
 
+    def _validate_config(self):
+        if self.name == "":
+            openai_model_list = sorted([m.id for m in self.client.models.list().data])
+            raise ValueError(
+                f"Model name is required for {self.generator_family_name}, use --model_name\n"
+                + "  API returns following available models: ▶️   "
+                + "  ".join(openai_model_list)
+                + "\n"
+                + "  ⚠️  Not all these are text generation models"
+            )
```

Comment:
Nice, that sounds about right. I started implementing a validation pattern here, as you can see, though it's a little harder to tease out for this module since `openai` is expected to be loaded in order to provide a helpful message. Will move this bit of validation (for when `name=""`) into `_load_client`.

Reply:
done in c0f17ed6e8cc15f67916747fdeab6d3622b63dc0